File: mobile/app/src/main/java/com/high5/models/user/UserDao.java
package com.high5.models.user;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

@Dao
public interface UserDao {

    @Query("SELECT * FROM user WHERE uid=(SELECT uid FROM user LIMIT 1) LIMIT 1")
    User getUser();

    @Query("UPDATE user set auth_token=(:token) where uid=(SELECT uid FROM user LIMIT 1)")
    void updateToken(String token);

    @Insert
    void insertAll(User... users);

    @Delete
    void delete(User user);
}

File: mobile/app/src/main/java/com/bdpsolutions/highfive/models/login/model/UserDao.kt
package com.bdpsolutions.highfive.models.login.model

import androidx.room.*

@Dao
interface UserDao {
    @Query("SELECT * FROM user LIMIT 1")
    fun getUser(): User?

    @Query("UPDATE user SET auth_token=(:token) WHERE uid=(SELECT uid from user LIMIT 1)")
    fun updateToken(token: String)

    @Insert
    fun addUser(user: User)
}
